low-level/files/opt/seagate/sspl/setup/sspl_test.py

```diff
@@ -38,31 +39,37 @@ def __init__(self, args: list):
         self.plan = "self_primary"
         self.avoid_rmq = False
         self.dbus_service = DbusServiceHandler()
-        # Load global, sspl and test configs
-        Conf.load(SSPL_CONFIG_INDEX, sspl_config_path)
-        Conf.load(SSPL_TEST_CONFIG_INDEX, sspl_test_config_path)
-        # Take copy of global config passed to sspl_test
         if args.config and args.config[0]:
-            sspl_test_gc_url = args.config[0]
+            self.sspl_test_gc_url = args.config[0]
         else:
-            sspl_test_gc_url = global_config_path
-        sspl_test_gc_copy_file = "/etc/sspl_test_gc_url.yaml"
-        with open(sspl_test_gc_copy_file, "w") as f:
-            f.write("")
-        self.sspl_test_gc_copy_url = "yaml://%s" % sspl_test_gc_copy_file
-        Conf.load(SSPL_TEST_GLOBAL_CONFIG, self.sspl_test_gc_copy_url)
-        Conf.load("global_config", sspl_test_gc_url)
-        Conf.copy("global_config", SSPL_TEST_GLOBAL_CONFIG)
+            self.sspl_test_gc_url = global_config_path
+        self.sspl_test_gc_copy_file = "/etc/sspl_test_gc_url.yaml"
 
     def validate(self):
         """Check for required packages are installed."""
+        # RPM dependency
+        rpm_deps = {
+            "cortx-sspl-test": None
+            }
         # python 3rd party package dependency
         pip3_3ps_packages_test = {
             "Flask": "1.1.1"
             }
         pkg_validator = PkgV()
-        pkg_validator.validate_pip3_pkgs(host=None, pkgs=pip3_3ps_packages_test,
-            skip_version_check=False)
+        pkg_validator.validate_pip3_pkgs(host=socket.getfqdn(),
+            pkgs=pip3_3ps_packages_test, skip_version_check=False)
+        pkg_validator.validate_rpm_pkgs(host=socket.getfqdn(),
+            pkgs=rpm_deps, skip_version_check=True)
```

Comment:
Until this changes are fine.

Why changes from line 64-72 needed?

Reply:
sspl_test/conf/sspl_test.conf won't be available until sspl-test rpm installed. So loading sspl_test.conf at init will throw error. So I moved out config loading part of code to validate method.

Reply:
When sspl-test is not installed, but user gets error at `conf.load("test_config", sspl_test.conf)` at init stage of sspl_test. At immediate sspl_test cmd execution, now sspl-test pkg validation happens first and then test conf loading is going to happen.

Reply:
Got it